packages/legend-query-builder/src/components/explorer/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -593,18 +663,69 @@ export const QueryBuilderPropertySearchPanel = observer(
                 <ResizablePanelSplitterLine color="var(--color-dark-grey-200)" />
               </ResizablePanelSplitter>
               <ResizablePanel>
-                <div className="query-builder-property-search-panel__results">
-                  {propertySearchPanelState.filteredPropertyNodes.map(
-                    (node) => (
-                      <QueryBuilderTreeNodeViewer
-                        key={node.id}
-                        node={node}
-                        queryBuilderState={queryBuilderState}
-                        level={1}
-                        stepPaddingInRem={0}
-                        explorerState={queryBuilderState.explorerState}
+                <div ref={searchModeRef}></div>
+                <BasePopover
+                  open={Boolean(propertySearchPanelState.isSearchPanelTipsOpen)}
+                  anchorEl={searchModeRef.current}
+                  onClose={() =>
+                    propertySearchPanelState.setIsSearchPanelTipsOpen(false)
+                  }
+                  anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right',
+                  }}
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right',
+                  }}
+                >
+                  <Panel>
+                    <div className="query-builder-property-search-panel__search__mode">
+                      <div className="query-builder-property-search-panel__form__section__header__label">
+                        Search Mode
+                        <button
+                          className="virtual-assistant__search__question__icon"
+                          tabIndex={-1}
+                          onClick={getDocumentationToImplementSvp}
+                        >
+                          <QuestionCircleIcon />
+                        </button>
+                      </div>
+                      <BaseRadioGroup
+                        className="query-builder-property-search-panel__search__mode--radio-group"
+                        value={propertySearchPanelState.modeOfSearch}
+                        onChange={handleSearchMode}
+                        row={false}
+                        options={propertySearchPanelState.modeOfSearchOptions}
+                        size={1}
                       />
-                    ),
+                    </div>
+                  </Panel>
+                </BasePopover>
+                {propertySearchPanelState.searchState.isInProgress && (
+                  <PanelLoadingIndicator isLoading={true} />
```

Comment:
Did we make this loader behave correctly?

Reply:
After addressing comment and removing the other unnecessary `this.searchState.inProgress();` I believe now this is proper 🤔